### DISCUSSION TITLE
Circle Style

### DISCUSSION THREAD

**Discussion by Saltssaumure** [Saltssaumure/fnf-osu-mania-skin#22] [General]
# What's in this update? #
- Added circle style osu!mania keys.
![image](https://user-images.githubusercontent.com/29710355/130361748-7ab3c9f9-816e-43c2-8ee3-77ad5d6c17dc.png)
- Cleaned up a few file names for clarity.

Every download contains all ten `Skin.ini` files in the `alternative-files` folder, so if you'd prefer to save space you don't need to download all ten `.osk`s to try out every version. Simply copy the desired `Skin.ini` into the skin's main folder.

# Plans for next update # 
Updates are based on user suggestions - this update was suggested by some YouTube comments! Leave your ideas here and they may be implemented next update.

<hr /><em>This discussion was created from the release <a href='https://github.com/Saltssaumure/fnf-osu-mania-skin/releases/tag/v1.7'>Circle Style</a>.</em>

**Comment by CalibriLight** [comment]
Go fail screen bf no blue balls
I can put bf from fnf assets without blue balls
Bf no blue balls assets here: https://gamebanana.com/skins/183665

**Reply by Saltssaumure** [reply]
Done, just added: https://github.com/Saltssaumure/fnf-osu-mania-skin/tree/master/alternative-files/textures/fail%20screen%20no%20balls

**Reply by Drago-Cuven** [reply]
> Done, just added: https://github.com/Saltssaumure/fnf-osu-mania-skin/tree/master/alternative-files/textures/fail%20screen%20no%20balls

blue pixel arrow squishes when pressed. distracts me and makes me screw up.

**Reply by Saltssaumure** [reply]
Thanks for letting me know, @Martinelplayz - will fix when I have time.

**Comment by geumsan** [comment]
A smaller kind of arrows. Y'know. The usual size that every skin uses.

Full size is too big, Mini is too small.

**Reply by Saltssaumure** [reply]
Will consider!

**Reply by Gavin123-code** [reply]
How do I make It downscroll for all mania K modes whatever you call it like 9k 8k 7k 6k 5k stuff like that

**Reply by ImRami491** [reply]
> How do I make It downscroll for all mania K modes whatever you call it like 9k 8k 7k 6k 5k stuff like that

go to the 5k options and lock the first option that comes up (below the key configuration) then do the same with 6k, 7k, 8k and 9k. in my case i only use upscroll

**Comment by Rebirthd** [comment]
Camellia arrows???

**Reply by Saltssaumure** [reply]
These won't be added for 2 reasons:

1. First and most important, the creator(s) of the vs Camellia mod do not give permission for others to use and distribute their assets.
2. The arrows themselves are based on Stepmania / DDR arrows, which already exist in many other osu! skins.

**Comment by JeffryCraft** [comment]
Hello, can you tell me how to remove the brightness of the arrows when pressing them? is that it annoys me a lot

**Reply by Saltssaumure** [reply]
If you mean the glow that appears behind the arrow, go into `[FNF skin folder]/mania/stage` and replace `stage-light.png` and `stage-light@2x.png` with a transparent png file. You could copy and rename `stage-hint.png` for instance.

**Comment by 2007MDFK** [comment]
why i cant put in downscroll?
Even i try invert it 
can you help me?

**Reply by 2007MDFK** [reply]
but its only happens on more than 4k xd

**Reply by Saltssaumure** [reply]
Try the solutions helpfully provided in #4

**Comment by OSUpleierButZurdo890** [comment]
when are you going to do the skin but with bars

**Reply by Saltssaumure** [reply]
It will be in the next update, but the timeline for the next update is not set yet. Generally I can work on side projects outside of term time, so mid-late December would be most likely. Watch this repository if you'd like to receive info about future updates.

**Comment by Starling1431** [comment]
Hi, i made some versions from your skin, the one is the HD/SHADED or Redrawned version, i made some "better" details and the other is the "better" pixel version that have more pixel assets.

Pixel version features:

*Pixel Count

*Pixel Bar and dead BF

*Blue Shaggy arrow fix

*Pixel mouse

and some assets.

HD/Redrawned:
*Taiko And mania totally shaded.

*Shaded Arrows

*Shaded Mouse and much more assets

Dowloads:

Pixel: https://www.mediafire.com/file/o7vrkyratrypp1s/FNF+v1.7+More+Better+Shaggy's+Pixel+Fullsize.osk/file

Redrawned: https://www.mediafire.com/file/8076oqeuuqbxydw/FNF+v1.7+Redrawned's+Fullsize.osk/file

**Reply by Saltssaumure** [reply]
Hi, good work on this edit! Have you considered making a fork of the original? That way, you can choose to easily incorporate the original's future updates. Plus people looking for the latest version of your skin will have one singular place to go, rather than having to find the most recent Mediafire link.

**Reply by Starling1431** [reply]
Ok you mean that instead of making links to mediafire make a link to this page and find the updates of my edit?  Well of course I will!  This way you will have more support and it will be easier for me.  Oh yes and could you make a video of my Skins as a plus in your videos?  Oh yes I also did a small update where I included the ring of sonic.exe (in triple trouble) for 5k down and I also made more customs logos for the modes

Sonic & Shaggy Link:

https://www.mediafire.com/download/avuxesvmfnnvrqs

**Comment by ChucrutconMayo** [comment]
hello good skin, in the future it would be possible to add the W A S D of the bob and bosip mod obviously when you have time and motivation (sorry for my bad English)  Very good skin :D

**Reply by Saltssaumure** [reply]
Hi, thank you for your kind feedback! I'm glad to say that osu! already
allows you to change your key binds to WASD. In osu!, open `Options`, then go to
`osu!mania layout` where you will find the option to change the keys.

**Reply by ChucrutconMayo** [reply]
I didn't mean that, I meant the bob and bosip mod skin lol, you can search on youtube what i mean

**Reply by Starling1431** [reply]
I think i can make the versión! U mean from the amor arrows right? Then i gonna make the skin!

**Reply by ChucrutconMayo** [reply]
you're right, wow thank you very much :D

**Comment by ImRami491** [comment]
could you make the 9k arrows bigger? small arrows make it hard for me to see and I can't see the patterns :(

**Reply by Saltssaumure** [reply]
The current fullsize version would cause 9K and above to clip into the score display on 16:9 resolutions. Fortunately there are two options:

1. If the clipping is not an issue for you, find the 9K section in the `Skin.ini` file and change references of `/keysmini/` to `/keys/`, which should give you the full size arrows.
2. I've also been working on a medium size arrow set, which will be about 75% scale. It is still subject to changes before release, but should be usable in its current state. [Download here](https://github.com/Saltssaumure/fnf-osu-mania-skin/archive/refs/heads/master.zip) and manually unzip into your skins folder, then copy a Midsize `Skin.ini` from the `alternative-files` folder into the main folder.

**Reply by ImRami491** [reply]
hello again, how can I make the arrows are right in the middle of the screen?

**Reply by Saltssaumure** [reply]
In the Skin.ini, change the value of `ColumnStart` for whichever keymodes you use. 
Decreasing the `ColumnStart` number will move the arrows closer to the left side of the screen.
Increasing the `ColumnStart` number will move the arrows closer to the right side of the screen.

**Reply by ImRami491** [reply]
tysm :) here are the numbers if anyones needs them (midsize):
4k: 290
5k: 250
6k: 210
7k: 200
8k: 150
9k: 70

**Comment by ChucrutconMayo** [comment]
Hi do you have plans for a future update?

**Reply by Saltssaumure** [reply]
At the moment, I'm happy with the skin's current features. If I have ideas or receive suggestions that would improve quality of life for a large group of users, then things may change.
